### PR TITLE
implement search API

### PR DESCRIPTION
**Pull request by MartinNowak**
- returns {name, description, latest_version} tuples

### CODE REVIEW

File: source/dubregistry/api.d [hunk 1]
```diff
@@ -17,6 +17,8 @@ void registerDubRegistryWebApi(URLRouter router, DubRegistry registry)
 }
 
 interface IPackages {
+	Json querySearch(string q);
```
Comment (s-ludwig) [inline source/dubregistry/api.d]: I'd prefer a strongly typed (and using `@method`) signature like in #115:
`@method(HTTPMethod.GET) string[] search(string q, uint maxResults=5);` (not sure about the `maxResults` parameter)

Having `Json` in the D API is always prone to silent breakage and field access bugs.

**Comment by s-ludwig** [comment]
Should we return the full package information for each package, or rather just the package name, or maybe something in-between (such as name + description + latest_version)? The full info could waste a lot of bandwidth for common scenarios.

**Comment by MartinNowak** [comment]
I don't think the full info is useful as a search result.

File: source/dubregistry/api.d (hunk 1 above)
Comment (MartinNowak) [inline source/dubregistry/api.d]: Will do.

Comment (MartinNowak) [inline source/dubregistry/api.d]: Why isn't there a `@GET` abbreviation for `@method(HTTPMethod.GET)`?

**Comment by MartinNowak** [comment]
Updated as requested.

**Comment by s-ludwig** [comment]
LGTM. I was thinking about allowing `@method!"GET"` as a shortcut. `@GET` would of course be a lot better still in that regard, but using it is usually rather rare, so it doesn't have to be as concise as possible (at the cost of polluting the global namespace).

**Pull request merged**